Character.cpp

```diff
@@ -1,117 +1,133 @@
+//Character.cpp
 #include "Character.h"
 #include <iostream>
 #include <vector>
 
 #include "DefensiveItem.h"
 #include "HelpfulItem.h"
+#include "Utility.h"
 
-Character::Character(int hp, int armor_, int attackDamage_ ) :
-    hitPoints(hp),
-    armor(armor_),
-    attackDamage(attackDamage_)
+Character::Character(int hp, int armor_, int attackDamage_) : 
+hitPoints(hp), armor(armor_), attackDamage(attackDamage_) 
 {
-    initialHitPoints.reset( new int(hitPoints) );
-    initialArmorLevel.reset( new int( armor) );
-    initialAttackDamage.reset( new int( attackDamage) );
+    initialHitPoints.reset(new int(hitPoints));
+    initialArmorLevel.reset(new int(armor));
+    initialAttackDamage.reset(new int(attackDamage));
 }
 
-void Character::attack( Character& other )
+int Character::takeDamage(int damage) 
 {
-    if( hitPoints <= 0 )
+    std::cout << getName() << " is taking " << damage << " damage!" << std::endl;
+    if (damage <= armor) 
+    {
+        armor -= damage;
+    } 
+    else 
+    {
+        damage -= armor;
+        armor = 0;
+    
+        hitPoints -= damage;
+        if (hitPoints < 0) 
+        {
+            std::cout << getName() << " has died in battle!" << std::endl;
+            hitPoints = 0;
+        }
+    }
+    
+  printStats();
+  return hitPoints;
+}
```

Comment:
This function did not need to be moved here.
This move has resulted in extra code needing to be reviewed that did not need reviewing.

Don't change code that does not need to be changed.

- [ ] revert

Reply:
understood